modules/analyzers/scancode-analyzer/main.go

```diff
@@ -147,122 +145,64 @@ func scancodeExitHandler(err error, output []byte) {
 	}
 }
 
-func readScancodeFile(resultfile string) (interface{}, error) {
-	re := regexp.MustCompile("{.+")
+func (scanalyzer *ScancodeAnalyzer) readScancodeFile(resultfile string) error {
 	scanResult, err := ioutil.ReadFile(resultfile)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("Error while reading the scancode input file: %v", err)
 	}
-	jsonScanResult := re.Find(scanResult)
-	var scanData interface{}
-	err = json.Unmarshal(jsonScanResult, &scanData)
+	err = json.Unmarshal(scanResult, &scanalyzer.Paths)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("Failed while unmarshalling json file: %v", err)
 	}
-	return scanData, nil
+	return nil
 }
 
-func scancode(workdir string, jobs int, resultfilepath string) (interface{}, error) {
+func (scanalyzer *ScancodeAnalyzer) scancode(workdir string, jobs int, resultfilepath string) error {
 	if resultfilepath == "" {
 		tmpfile, err := ioutil.TempFile("", "qmstr-scancode")
 		resultfilepath = tmpfile.Name()
 		if err != nil {
-			return nil, err
+			return err
 		}
 		defer os.Remove(tmpfile.Name())
 	}
 
-	cmdlineargs := []string{"--full-root", "-l", "-c", "--json", resultfilepath}
+	cmdlineargs := []string{"--full-root", "-l", "-c", "--custom-output", resultfilepath, "--custom-template", "qmstr.j2"}
```

Comment:
Note that this `--custom-output` option is a slightly less traveled area, so there may be some dragons here and there... and there are some things that you cannot get from that such as the detected license texts (IMHO you should use the `--license-text` ) 
Also I see that you may be missing the --package option to collect package manifests and their licenses.

Reply:
Thanks for the input @pombredanne! So your opinion is to skip the `--custom-output` flag? You believe that we are going to miss some license information when using the flag?